### ISSUE TITLE
toolchain: python toolchain has duplicated entries for system (macOS) python 

### ISSUE THREAD

**Issue by pantheraleo-7** [zed-industries/zed#21679]
### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

IIRC this wasn't the case when toolchain first landed. So it's probably a regression.

### Environment

Zed: v0.165.3 (Zed Preview)
OS: macOS 15.1.1
Memory: 8 GiB
Architecture: aarch64

### If applicable, add mockups / screenshots to help explain present your vision of the feature

<img width="546" alt="Screenshot 2024-12-07 at 1 46 18 PM" src="https://github.com/user-attachments/assets/0c230cdd-e18e-47d0-bf88-10f9a96c42a3">

### If applicable, attach your Zed.log file to this issue.

<details><summary>Zed.log</summary>

```
[Zed.log](https://github.com/user-attachments/files/18047676/Zed.log)

```
</details>

**Comment by pantheraleo-7** [comment]
Also, the UX is quite confusing? The first entry is always highlighted when the toolchain selector is reopened irrespective of the python installation in use. @danilo-leal

**Comment by danilo-leal** [comment]
@pantheraleo-7 thanks for the feedback! The first item of the list being "selected" (technically, focused) isn't something specific to this use case; it's how the picker component behaves. For example, if you open the command palette (cmd + shift + p), or the file picker (cmd + p), you'll see the same thing, too. That's so you can traverse the list up and down with your keyboard. 

However, there may be an improvement opportunity in this specific scenario where the first item could potentially be the installation you're currently using?

**Comment by osiewicz** [comment]
Thanks for reporting this issue, I've just merged a fix.

**Comment by pantheraleo-7** [comment]
> if you open the command palette (cmd + shift + p), or the file picker (cmd + p), you'll see the same thing, too.

In the case of file picker, the active tab seems to be always the first entry. The command palette is sorted alphabetically. So these make sense. And these aren't "selectors" like the toolchain.

> However, there may be an improvement opportunity in this specific scenario where the first item could potentially be the installation you're currently using?

Yes! Because it gets really confusing when two items have the same name, and it is quite common for venvs to have the same name.

I had the first and second entry with the same name, and I selected the second entry 3-4 times because I thought it wasn't getting selected. Then I ran some code to confirm that it was indeed getting selected haha.
